static/assets.js

```diff
@@ -36,4 +36,47 @@ function hideEmptySections() {
         }
         section.style.display = 'none'
     })
-}
\ No newline at end of file
+}
+
+//  ------------    Version Filtering
+function normalize_version(raw_version) {
+    let version = raw_version
+        .replace(/^[^\d]+/, '')
+        .replace(/[^\d]+$/, '');
+    return version ? Array.from({ ...version.split('.'), length: 3 }, (v, i) => v ?? 0).join('.') : null;
+}
+
+let versionsQuery = document.querySelectorAll('.asset-card .asset-card__tags .asset-card__bevy-versions .asset-card__tag');
+let versions = [...new Set([...versionsQuery]
+    .map(item => normalize_version(item.innerText))
+    .filter(i => i)
```

Comment:
I would rather prefer if the list was sorted based on the version.
My js is a bit rusty, but this seems to work for me:
![image](https://github.com/bevyengine/bevy-website/assets/25123512/c9231978-f6f8-40d7-a3f9-1000368c6f7f)

```suggestion
    .filter(i => i)
    .sort((a, b) => {
        let score = 0;
        let av = a.split('.');
        let bv = b.split('.');
        let ln = av.length < bv.length ? av.length : bv.length;
        for (let index = 0; index < ln-1; index++) {
            score += (parseInt(av[index]) - parseInt(bv[index])) * (ln - index);
        }
        return score;
    }).reverse()
```

Reply:
It may need a comment or two 😅

Reply:
from what I see, it does not sort by minor version

Reply:
It should be, though, it might not work if its "0.4" vs "0.4.1" , you can try to insert additional minor versions in normalize_version or the like

Reply:
There were some unhandled edge cases for patch, someone did it better:
https://medium.com/geekculture/sorting-an-array-of-semantic-versions-in-typescript-55d65d411df2
```suggestion
    .filter(i => i)
    .sort((a, b) => {
        // 1. Split the strings into their parts.
        let a1 = a.split('.');
        let b1 = b.split('.');
        // 2. Contingency in case there's a 4th or 5th version
        const len = Math.min(av.length, bv.length);
        // 3. Look through each version number and compare.
        for (let i = 0; i < len; i++) {
            const a2 = +a1[ i ] || 0;
            const b2 = +b1[ i ] || 0;
            if (a2 !== b2) {
                return a2 > b2 ? 1 : -1;        
            }
        }
        // 4. We hit this if the all checked versions so far are equal
        return b1.length - a1.length;
    }).reverse()
```

Reply:
Heres with extra test versions: 
![image](https://github.com/bevyengine/bevy-website/assets/25123512/1e96b346-c264-45b3-a92c-63bcc248d328)

Reply:
optimised this a bit